src/Npgsql/NpgsqlRawCopyStream.cs

```diff
@@ -193,7 +195,9 @@ public void Cancel()
                 _isDisposed = true;
                 _writeBuf.EndCopyMode();
                 _writeBuf.Clear();
-                _connector.SendMessage(new CopyFailMessage());
+                _connector.WriteCopyFail();
+                _connector.WriteCopyFail();
```

Comment:
Duplicate?

Reply:
Oops!

Reply:
I was a little surprised that this passed the test suite. Should we add a test covering this path? Or is one of our preexisting tests just missing this?

Reply:
@roji Not sure if you saw this (due to the force-push). Should this path be covered? Or did the double-write not fail for some other reason?

Reply:
Yeah, missed your previous comment...

I'm pretty sure this path is actually covered by `CancelRawBinaryImport`, but this specific mistake may not show up there, as the server really is supposed to respond to the cancellation with an error (not 100% sure though)